src/Content/Version.php

```diff
@@ -125,61 +163,74 @@ public function modified(
 	public function move(string $fromLanguage, VersionId $toVersionId, string $toLanguage): void
 	{
 		$this->ensure($fromLanguage);
-		$this->model->storage()->move($this->id, $fromLanguage, $toVersionId, $toLanguage);
+		$this->model->storage()->move(
+			fromVersionId: $this->id,
+			fromLanguage: $this->language($fromLanguage),
+			toVersionId: $toVersionId,
+			toLanguage: $this->language($toLanguage)
+		);
 	}
 
 	/**
 	 * Returns the stored content fields
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 * @return array<string, string>
 	 */
 	public function read(string $language = 'default'): array
 	{
 		$this->ensure($language);
-		return $this->model->storage()->read($this->id, $language);
+		return $this->model->storage()->read($this->id, $this->language($language));
 	}
 
 	/**
 	 * Updates the modification timestamp of an existing version
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
+	 * @param string|null $language If null, all available languages will be touched
 	 *
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
 	public function touch(string|null $language = null): void
 	{
-		// touch all languages
-		if ($language === null) {
-			foreach ($this->model->kirby()->languages() as $language) {
-				$this->touch($language->code());
-			}
+		// touch a single language
+		if ($this->model->kirby()->multilang() === false) {
+			$this->touchLanguage('default');
+			return;
 		}
 
-		// make sure the version exists
-		$this->ensure($language);
-
-		// touch the default language in single-language mode
-		if ($this->model->kirby()->multilang() === false) {
-			$this->model->storage()->touch($this->id, 'default');
+		// touch a specific language
+		if ($language !== null) {
+			$this->touchLanguage($language);
 			return;
 		}
 
-		// touch a single language
-		$this->model->storage()->touch($this->id, $language);
+		// touch all languages
+		foreach ($this->model->kirby()->languages() as $language) {
+			$this->touchLanguage($language);
+		}
+	}
+
+	/**
+	 * Updates the modification timestamp of a specific language
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
+	 */
+	public function touchLanguage(string $language = 'default'): void
```

Comment:
`Question`: What were your thoughts about the separate `touchLanguage` method? From what I can tell when reading the code, the logic here could be integrated into `touch()` directly, but I'm curious about your plans and thoughts.

Reply:
It's an artefact from when we still had ::deleteLanguage to keep both methods (delete and touch) consistent. When I removed ::deleteLanguage I was wondering if I should delete this method as well and the decided against it for some reason that I cannot remember. I think it really isn't needed at all.

Reply:
`Correction`: I've looked at ways to remove `::touchLanguage` and it would turn `::touch` into a lot more Spaghetti code to remove it. I decided to keep it but turned it into a protected method to be used as an internal helper for us. 

`Question`: I think the bigger decision here is if we want to keep `null` as option to touch all versions. You had that question before and I'm still not 100% sure about it. 

`Suggestion:` I would keep it as it is for now. But maybe we can turn it into a todo/question for later in Notion.

Reply:
You are right. I removed touchLanguage and simplified touch by removing the null option.